### PR TITLE
Implement log scale of parallel coordinate

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

NA

Log axis if distribution is log scale for parallel coordinate plot

## scripts
```
import logging

import optuna
from optuna_dashboard._app import create_app


host = "127.0.0.1"
port = 8080

optuna.logging.set_verbosity(logging.CRITICAL)


def create_optuna_storage() -> optuna.storages.InMemoryStorage:
    storage = optuna.storages.InMemoryStorage()

    # Single-objective study
    study = optuna.create_study(study_name="single-objective", storage=storage)

    def objective_single(trial: optuna.Trial) -> float:
        x1 = trial.suggest_float("x1", 0, 10)
        x2 = trial.suggest_float("x2", 0.1, 10, log=True)
        x3 = trial.suggest_categorical("x3", ["foo", "bar"])
        return (x1 - 2) ** 2 + (x2 - 5) ** 2

    study.optimize(objective_single, n_trials=100)

    return storage


def main() -> None:
    storage = create_optuna_storage()
    app = create_app(storage, debug=True)
    app.run(host=host, port=port, reloader=True)


if __name__ == "__main__":
    main()
```

### CODE REVIEW

File: optuna_dashboard/ts/components/GraphParallelCoordinate.tsx [hunk 1]
```diff
@@ -169,6 +169,20 @@ const plotCoordinate = (
...
+    })
+    const minValue = Math.min(...logValues)
+    const maxValue = Math.max(...logValues)
+    const tickvals = Array.from(
+      { length: Math.ceil(maxValue) - Math.floor(minValue) + 1 },
+      (_, i) => i + Math.floor(minValue)
+    )
+    const ticktext = tickvals.map((x) => `${Math.pow(10, x).toPrecision(3)}`)
+    return { logValues, tickvals, ticktext }
+  }
```
Comment (keisuke-umezawa) [inline optuna_dashboard/ts/components/GraphParallelCoordinate.tsx:184]: ref: https://github.com/optuna/optuna/blob/master/optuna/visualization/_parallel_coordinate.py#L213-L230

**Comment by c-bata** [comment]
@not522 @gen740 Could you review this PR?

**Review by gen740 — APPROVED** [review]
LGTM! It works as expected.

File: optuna_dashboard/ts/components/GraphParallelCoordinate.tsx [hunk 2]
```diff
@@ -169,6 +169,21 @@ const plotCoordinate = (
       .join("")
   }
 
+  const calculateLogScale = (values: number[]) => {
+    const logValues = values.map((v) => {
+      return Math.log10(v)
+    })
+    const minValue = Math.min(...logValues)
+    const maxValue = Math.max(...logValues)
+    const range = [minValue, maxValue]
```
Comment (not522) [inline optuna_dashboard/ts/components/GraphParallelCoordinate.tsx:178]: Both ends of ticks should be within the range.
```suggestion
    const range = [Math.floor(minValue), Math.ceil(maxValue)]
```

**Review by not522 — COMMENTED** [review]
Thank you for your PR. Please check my comment.

**Comment by keisuke-umezawa** [comment]
@not522 Thank you for reviewing. Could you check it again?

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**